src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Migrations/20240501113006_InitialMigration.cs

```diff
@@ -579,40 +580,67 @@ protected override void Up(MigrationBuilder migrationBuilder)
                 principalTable: "DataSetVersions",
                 principalColumn: "Id");
 
+            var isLocalEnvironment = 
+                Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == Environments.Development;
+
             // Grant privileges on objects created by this resource's database user to the Admin App Service user.
-            var adminAppServiceIdentityName = Environment.GetEnvironmentVariable("AdminAppServiceIdentityName");
-            if (adminAppServiceIdentityName != null)
+            var adminAppServiceRoleName = isLocalEnvironment
+                ? "app_admin" 
+                : Environment.GetEnvironmentVariable("AdminAppServiceIdentityName");
+            if (adminAppServiceRoleName != null)
             {
                 migrationBuilder.Sql(
-                    $"GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO \"{adminAppServiceIdentityName}\"");
+                    $"""GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO {adminAppServiceRoleName}""");
```

Comment:
I wondered if you were going to go with an approach of defining roles up front with their associated permissions to do `SELECT, INSERT, UPDATE, DELETE`, and then assign the Azure users those roles.  Had you looked at that approach at all?

E.g. every environment always has an "app_admin" role with NOLOGIN and `SELECT, INSERT, UPDATE, DELETE` permissions, and then we assign the Azure Admin managed identity user that role, and locally we assign a db user that role.

Or alternatively we have a `public_api_read_write` role that is NOLOGIN with `SELECT, INSERT, UPDATE, DELETE` permissions, and then we just assign all our application users that role.

There's possibly not much advantage in either alternative approach to be fair though, if they even work.

Reply:
Discussed this on Slack and explained that I'd initially tried something like this but couldn't get it to work unless the Public Data API's app user which creates and owns all the objects also did the grants.

I think I've got this working now but done this as a separate commit so we can revert back if necessary.

I went with this approach:

> Or alternatively we have a public_api_read_write role that is NOLOGIN with SELECT, INSERT, UPDATE, DELETE permissions, and then we just assign all our application users that role.

We would assign this role to all our application user roles and also our own Azure admin user roles that we want to have write privileges.

Reply:
Cool!